src/app/components/pool-size-chart/pool-size-chart.component.ts

```diff
@@ -31,10 +31,66 @@ export class PoolSizeChartComponent implements OnInit {
   ) {}
 
   ngOnInit(): void {
-    this.setupChart();
+    this.config = {
+      curveType: 'function',
+      legend: 'none',
+      backgroundColor: 'transparent',
+      colors: [this.palette.series],
+      chartArea: {
+        width: '90%',
+        height: '80%',
+        right: 0,
+        left: 90,
+      },
+      hAxis: {
+        viewWindowMode: 'maximized',
+        textStyle: {
+          color: this.palette.text,
+        },
+        gridlines: {
+          color: this.palette.gridlines,
+          count: 1,
+        },
+        minorGridlines: {
+          color: this.palette.gridlines,
+          count: 0,
+        },
+      },
+      vAxis: {
+        logScale: false,
+        viewWindowMode: 'maximized',
+        format: '#.#TB',
+        textStyle: {
+          color: this.palette.text,
+        },
+        gridlines: {
+          color: this.palette.gridlines,
+        },
+        minorGridlines: {
+          count: 0,
+        }
+      },
+      pointSize: 3,
+    };
+    
+    this.setupChart(this.config);
+  }
+
+  ngOnChanges(changes : any) {
+    if ( changes.config != undefined ) {
+      this.config = changes.config.currentValue;
+      this.setupChart(this.config);
+    }
+  }
+
+  setLogScale(logScale: Boolean): void {
+    this.config.vAxis.logScale = logScale;
+    this.setupChart(this.config);
+    this.linearScaleButtonElement.nativeElement.disabled = !logScale;
+    this.logScaleButtonElement.nativeElement.disabled = logScale;
```

Comment:
Better to have a class member called `logScale` and set button state in the template <button [disabled]="logScale">.